addons/website_sale_delivery/static/src/js/website_sale_delivery.js

```diff
@@ -35,24 +39,207 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
+        }
+    },
+    /**
+     * @private
+     * @param {Object} doc_carrier //carrier element from document
+     */
+    _specificDropperDisplay: function (doc_carrier) {
+        if(!doc_carrier)
+            return;
+        if(!doc_carrier.closest("li").getElementsByTagName("input")[0].attributes.getNamedItem("delivery_type"))
+            return;
+        let current_carrier_type = doc_carrier.closest("li").getElementsByTagName("input")[0].attributes.getNamedItem("delivery_type").nodeValue;
+        const location_still_loading = document.getElementById("#"+current_carrier_type+"_addresses")
+        if(location_still_loading)return
+        while (doc_carrier.firstChild) {
+            doc_carrier.lastChild.remove();
+        }
+        const current_carrier_checked = doc_carrier.closest("li").getElementsByTagName("input")[0].checked;
+        const span = document.createElement("em");
+        const is_pick_up_location_list = doc_carrier.nextElementSibling.childNodes.length > 0
+        if(current_carrier_checked){
+            const chevron_down = document.createElement("i");
+            if(is_pick_up_location_list)
+                chevron_down.classList.add("fa", "fa-angle-up");
+            else
+                chevron_down.classList.add("fa", "fa-angle-down");
+            span.textContent = "Pick-Up Locations ";
+            span.classList.add("link-primary");
+            span.appendChild(chevron_down);
+        }
+        else {
+            span.textContent = "select to see available Pick-Up Locations";
+            span.classList.add("text-muted");
+        }
+        doc_carrier.appendChild(span);
+    },
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onClickLocation: async function (ev) {
+        ev.preventDefault();
+        const carrierId = ev.currentTarget.closest(".o_delivery_carrier_select").childNodes[1].value;
+        await this._checkCarrier(ev,carrierId)
+        const self = this;
+        const modal = ev.target.closest(".o_list_pickup_locations");
+        const encoded_location = ev.target.previousElementSibling.innerText;
+        await self._rpc({
+            route: "/shop/access_point/set",
+            params: {
+                access_point_encoded: encoded_location,
+            },
+        })
+        while (modal.firstChild) {
+            modal.lastChild.remove();
+        }
+        await self._getCurrentLocation();
+        var $payButton = $('button[name="o_payment_submit_button"]');
+        var disabledReasons = $payButton.data('disabled_reasons') || {};
+        disabledReasons.carrier_selection = false;
+        $payButton.data('disabled_reasons', disabledReasons);
+        $payButton.prop('disabled', _.contains($payButton.data('disabled_reasons'), true));
+        var error_messages = $('#error_no_pick_up_point')
+        if (error_messages.length > 0){
+            for(let i = 0;i<error_messages.length;i++)
+            error_messages[i].remove();
+        }
+    },
```

Comment:
I guess this code comes from legacy code? Please modernize it (use const, no jquery)
The new equivalent to jquery's `.data` is as such: 
```js
Data.set(elem, key, value);
Data.get(elem, key);
```